crates/types/src/state_diffs.rs

```diff
@@ -59,3 +60,73 @@ where
         None => serializer.serialize_none(),
     }
 }
+
+impl ContractUpdate {
+    // Helper function to create a key for sorting
+    fn sort_key(&self) -> BigUint {
+        self.address.clone()
+    }
+
+    fn has_same_storage_updates(&self, other: &ContractUpdate) -> bool {
```

Comment:
this method should be implemented for storage updates, not ContractUpdate ideally. all structs can have `unordered_eq` method which can internally call other `unordered_eq` methods. And we can have an `UnorderedEq` trait which can be implemented by all.